### PR TITLE
Only enqueue installed packages eligible for upgrade on package updates

### PR DESCRIPTION
**Pull request by ewrenn8**

### CODE REVIEW

File: cmd/controller/handlers/installed_pkg_version_handler.go [hunk 1]
```diff
@@ -31,48 +35,69 @@ func NewInstalledPkgVersionHandler(c kcclient.Interface, log logr.Logger) *Insta
...
+	err := ipvh.enqueueEligibleInstalledPackages(q, evt.Object)
 	if err != nil {
 		ipvh.log.Error(err, "enqueueing all installed pakcages")
 	}
 }
 
-func (ipvh *InstalledPkgVersionHandler) enqueueAllPackages(q workqueue.RateLimitingInterface) error {
+func (ipvh *InstalledPkgVersionHandler) enqueueEligibleInstalledPackages(q workqueue.RateLimitingInterface, obj runtime.Object) error {
+	pkg := obj.(*pkgv1alpha1.Package)
 	installedPkgList, err := ipvh.client.InstallV1alpha1().InstalledPackages("").List(metav1.ListOptions{})
```
Comment (danielhelfand) [inline cmd/controller/handlers/installed_pkg_version_handler.go:70]: Could we use `meta.ListOptions{}` in any way here to only pull back installed package with a specific public name?

Comment (ewrenn8) [reply cmd/controller/handlers/installed_pkg_version_handler.go:70]: Currently, `metadata.name` and `metadata.namespace` are the only supported field selectors, so we aren't able to filter on the `spec.packageRef.publicName` or `spec.packageRef.version`

Comment (danielhelfand) [reply cmd/controller/handlers/installed_pkg_version_handler.go:70]: Gotcha, pulling everything back makes sense then.

**Review by danielhelfand — APPROVED** [review]
lgtm

**Pull request merged**